vowpalwabbit/example.h

```diff
@@ -88,11 +88,24 @@ struct example : public example_predict  // core example datatype.
   size_t example_counter = 0;
 
   // helpers
-  size_t num_features = 0;         // precomputed, cause it's fast&easy.
+  // precomputed, cause it's fast&easy. These values are the number of singular/non interacted features.
+  size_t num_features = 0;
+  // precomputed, cause it's fast&easy.
```

Comment:
where are these precomputed?

Reply:
Because they can be trivially precomputed in the parser, so there's no reason to do it at a later stage. I believe they are always used by the shared_data counters.